### PR TITLE
Sync to nodejs/master (2016-05-12)

### PR DESCRIPTION
**Pull request by kunalspathak**
##### Checklist

- [x] tests and code linting passes
- [x] a test and/or benchmark is included
- [x] the commit message follows commit guidelines
##### Affected core subsystem(s)

deps, tools
##### Description of change

Sync to nodejs/master as of (2016-05-12).

### CODE REVIEW

**Comment by kunalspathak** [comment]
cc: @nodejs/node-chakracore

File: test/parallel/test-console.js [hunk 1]
```diff
@@ -118,6 +116,8 @@ for (const expected of expectedStrings) {
   assert.equal(expected + '\n', errStrings.shift());   // console.warn  (stderr)
 }
 
+assert.equal(expectedFuncToString, strings.shift());
+assert.equal(expectedFuncToString, strings.shift());
```
Comment (jianchun) [inline test/parallel/test-console.js:120]: Why this change... was it just missing from merge?

Comment (kunalspathak) [inline test/parallel/test-console.js:120]: There was some refactoring in the test case and my merge deleted some of the lines. This is just regaining the chakracore version of test case.

Comment (jianchun) [inline test/parallel/test-console.js:120]: looks so

**Comment by jianchun** [comment]
LGTM, thanks!

**Comment by thefourtheye** [comment]
Looks like mostly linter error fixes. LGTM
